src/wallet/coin_selection.rs

```diff
@@ -296,16 +321,31 @@ fn select_sorted_utxos(
         .collect::<Vec<_>>();
 
     let amount_needed_with_fees = amount_needed + fee_amount;
+
     if selected_amount < amount_needed_with_fees {
         return Err(Error::InsufficientFunds {
             needed: amount_needed_with_fees,
             available: selected_amount,
         });
     }
 
+    let remaining_amount = selected_amount - amount_needed_with_fees;
+
+    let change_fee = fee_rate.fee_vb(serialize(drain_output).len());
+
+    let drain_val = remaining_amount.saturating_sub(change_fee);
+
+    let excess = if drain_val.is_dust(&drain_output.script_pubkey) {
+        let target = drain_output.script_pubkey.dust_value().as_sat();
```

Comment:
What is target, exactly? The target you should reach for creating a change?

Reply:
The idea was that you must _hit_, at least, the dust value to create change (_the target_) . Now it doesn't seem so accurate. Maybe `dust_threshold` is..

Reply:
Yeah `dust_threshold` is definitely easier to understand!